### PR TITLE
docs(release): add `v1.3.8` release details to CHANGELOG

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Add the `v1.3.8` release details to the CHANGELOG in preparation for the upcoming `v1.3.8` release.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,45 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+* Healthcheck definitions in Dockerfile ([#1134](https://github.com/openfga/openfga/pull/1134)). Thanks @Siddhant-K-code!
+
+### Changed
+* Database iterators yielded by the RelationshipTupleReader storage interface now accept a `context` parameter which allows iteration to be promptly terminated ([#1055](https://github.com/openfga/openfga/pull/1055))
+
+  We have noticed improvements in query performance by adding this because once a resolution path has been found we more quickly cancel any further evaluation by terminating the iterators promptly.
+
+* Normalized the in memory storage adapter implementation to make use of TupleRecords internally ([#1180](https://github.com/openfga/openfga/pull/1180))
+
+* Improved tuple validation peformance with precomputation of TTUs ([#1171]https://github.com/openfga/openfga/pull/1171)
```
Comment (jpadilla) [inline CHANGELOG.md:37]:
```suggestion
* Improved tuple validation peformance with precomputation of TTUs ([#1171]https://github.com/openfga/openfga/pull/1171))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,45 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+* Refactored the commands in the `pkg/server/commands` package to uniformly use the Options builder pattern ([#1142](https://github.com/openfga/openfga/pull/1142)). Thanks for the contribution @ilaleksin!
+
+* Upgraded to Go `1.21.4` ([#1143](https://github.com/openfga/openfga/pull/1143)). Thanks @tranngoclam!
+
+### Fixed
+* Cache key computation with contextual tuples with different ordering. ([#1187](https://github.com/openfga/openfga/pull/1187))
+
+  If two requests were made with the same request body and contextual tuples but the order of the contextual tuples differed, then the cache key that was produced would differ.
+
+* Missing child spans in OpenTelemetry traces ([#1196](https://github.com/openfga/openfga/pull/1196))
```
Comment (miparnisari) [inline CHANGELOG.md:48]:
```suggestion
* Use `NoOp` TracerProvider if tracing is disabled ([#1139](https://github.com/openfga/openfga/pull/1139) and [#1196](https://github.com/openfga/openfga/pull/1196))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,45 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+* Improved tuple validation peformance with precomputation of TTUs ([#1171]https://github.com/openfga/openfga/pull/1171))
+
+* Refactored the commands in the `pkg/server/commands` package to uniformly use the Options builder pattern ([#1142](https://github.com/openfga/openfga/pull/1142)). Thanks for the contribution @ilaleksin!
+
+* Upgraded to Go `1.21.4` ([#1143](https://github.com/openfga/openfga/pull/1143)). Thanks @tranngoclam!
+
+### Fixed
+* Cache key computation with contextual tuples with different ordering. ([#1187](https://github.com/openfga/openfga/pull/1187))
+
+  If two requests were made with the same request body and contextual tuples but the order of the contextual tuples differed, then the cache key that was produced would differ.
```
Comment (miparnisari) [inline CHANGELOG.md:46]:
```suggestion
* If two requests were made with the same request body and contextual tuples but the order of the contextual tuples differed, then the cache key that is produced is now the same.([#1187](https://github.com/openfga/openfga/pull/1187))

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,45 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+  Similar to the server's main entrypoint `openfga run`, you can now override the datastore username and password with environment variables. when running the `openfga migrate` utility.
+
+* Healthcheck definitions in Dockerfile ([#1134](https://github.com/openfga/openfga/pull/1134)). Thanks @Siddhant-K-code!
+
+### Changed
+* Database iterators yielded by the RelationshipTupleReader storage interface now accept a `context` parameter which allows iteration to be promptly terminated ([#1055](https://github.com/openfga/openfga/pull/1055))
+
+  We have noticed improvements in query performance by adding this because once a resolution path has been found we more quickly cancel any further evaluation by terminating the iterators promptly.
+
+* Normalized the in memory storage adapter implementation to make use of TupleRecords internally ([#1180](https://github.com/openfga/openfga/pull/1180))
```
Comment (miparnisari) [inline CHANGELOG.md:35]: Ehh I wouldn't mention this, I don't think end users will care about the internal implementation of the memory adapter.

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,45 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+
+* Improved tuple validation peformance with precomputation of TTUs ([#1171]https://github.com/openfga/openfga/pull/1171))
```
Comment (miparnisari) [inline CHANGELOG.md:37]:
```suggestion
* Improved tuple validation peformance with precomputation of TTUs ([#1171](https://github.com/openfga/openfga/pull/1171))
```

File: CHANGELOG.md (hunk 2 above)
Comment (jon-whit) [reply CHANGELOG.md:48]: That change doesn't describe what was fixed, it describes what was done for the fix. If I'm a developer I care about _why_ we did what we did, not how we did.

File: CHANGELOG.md (hunk 4 above)
Comment (miparnisari) [reply CHANGELOG.md:74]: I don't care :)

Comment (jon-whit) [reply CHANGELOG.md]: Removed.

**Review by jpadilla — APPROVED** [review]

**Review by miparnisari — APPROVED** [review]

**Pull request merged**